x/lightclient/keeper/hook_listener.go

```diff
@@ -41,9 +36,12 @@ func (hook rollappHook) AfterUpdateState(
 
 	client, ok := hook.k.GetCanonicalClient(ctx, rollappId)
 	if !ok {
-		client, ok = hook.k.GetProspectiveCanonicalClient(ctx, rollappId, stateInfo.GetLatestHeight()-1)
+		client, ok = hook.k.FindMatchingClient(ctx, stateInfo)
 		if ok {
 			hook.k.SetCanonicalClient(ctx, rollappId, client)
+			if err := hook.k.PruneSignersBelow(ctx, client, stateInfo.GetLatestHeight()+1); err != nil {
+				return errorsmod.Wrap(err, "prune signers")
+			}
 		}
 		return nil
 	}
```

Comment:
```suggestion
			
		}
		
	}
```

Removing the `return nil` : makes sure people are not stuck with unbond restriction

Removing: prune signers below: it's called below too, so no need